### ISSUE TITLE
YAML error

### ISSUE THREAD

**Issue by merima-d** [Watts-College/cpp-526-sum-2024#22]
It is utterly frustrating that by trying to knit or run R document results in the error below. When running ' Run all' all seem to be fine but the code is halted by the error. Does anybody know how to address it? And this message even appears on the previous assignments that used to run normally. I started having this problem with our final projects. Also, the libraries need to be constantly loaded almost each time I open the file . The other projects get attached to my current project automatically. I've been saving the files the way I usually do and do not understand if there was a different way of saving the files that could solve the problem.    

Error in yaml::yaml.load(..., eval.expr = TRUE) : 
  Scanner error: mapping values are not allowed in this context at line 7, column 20
Calls: <Anonymous> ... parse_yaml_front_matter -> yaml_load -> <Anonymous>
Execution halted

**Comment by CTNovoa** [comment]
Unfortunately, I am not sure. I am also experiencing an issue when I go to knit.

Even though I have the dplyr package installed and updated, R is saying that it cannot "find the function %>%" when I go to knit...

**Comment by JessiMT** [comment]
Hello Merima,

I encountered the same issue when I downloaded the template from the instructions file. However, when I reopened the template provided directly in the module (not the one inside the instructions file), it worked fine.

I wasn't able to find a proper solution to this error on forums, but I hope this workaround helps you too while we wait for the professor's response. This way, you can start working on the final dashboard.

Jessica

**Comment by ngipe7** [comment]
@CTNovoa  I found that if I removed pander() from the end of my code, this error would go away. I am unsure why this works.

**Comment by gmcirco** [comment]
Hi Merima,

On which lab or project are you encountering this error? The error listed looks like there is something wrong with your YAML header on top of the project. The header looks like this:

---
title: "Final Project: Practice Tab"
output: 
  flexdashboard::flex_dashboard:
    theme: spacelab
    source: embed
    smart: false
runtime: shiny
---

The error specifically is pointing to line 7, column 20 which indicates there might be something missing. Can you provide this?

**Comment by merima-d** [comment]
Thank you all for responding to my issue. I resolved it by starting anew from the module file. I just saw JessiMT message and I believe that the new version I've been successful with was one directly from the module and not from the one found in Instruction Template. After one day lost due to searching for help online, I decided to delete all files I put in different folders and directories on my PC as a result of my attempts to resolve the issue and opened the new one from the Module location. It worked well. 

The header was not the problem, -while all forums were talking about indentation issues- that was not it. The issue was that the files very 'moody.' One night they opened nicely and the next morning they would have the same YAML message. That 'bug' was also carried on the files I didn't run, for example Full dashboard that did work fine but after the message on the Isolated file it would not work . To note: The header in my initial post has lost the original format by pasting it here, however, it looks as same as posted by gmcirco on my files.
